static/js/favicon.js

```diff
@@ -52,14 +59,49 @@ export function update_favicon(new_message_count, pm_count) {
                 ? `${Math.floor(new_message_count / 1e3 ** pow)}${suffix}`
                 : "∞";
         const count_long = count.length > 2;
-        const rendered_favicon = render_favicon_svg({
-            count,
-            count_long,
-            have_pm: pm_count !== 0,
-            favicon_font_url,
-        });
 
-        load_and_set_favicon(rendered_favicon);
+        // we need to generate a data uri, we cannot just pass a href into
+        // the template because we generate an immutable blob from the
+        // template which would not include the rendered image
+        if (user_settings.realm_icon_as_favicon && page_params.realm_icon_url) {
+            if (!image_data_URI || should_update_image_data_URI) {
```

Comment:
`image_data_URI` → `icon_data_url` (`icon` is more specific, we use lowercase variable names, and “URI” is [not a thing anymore](https://url.spec.whatwg.org/#goals)).

Don’t use an explicitly managed `should_update_image_data_URI` flag. Instead, make it automatic: remember the `realm_icon_url` corresponding to the cached `icon_data_url`, and update them both if `page_params.realm_icon_url !== realm_icon_url`.